### PR TITLE
Bluetooth: Mesh: Add initial implementation

### PR DESCRIPTION
**Pull request by jhedberg**
Add an initial implementation for the Bluetooth Mesh Profile
Specification. The main code resides in subsys/bluetooth/host/mesh and
the public API can be found in include/bluetooth/mesh.h. There are a
couple of samples provided as well under samples/bluetooth and
tests/bluetooth.

The implementation covers all layers of the Bluetooth Mesh stack and
most optional features as well. The following is a list of some of
these features and the c-files where the implementation can be found:

 - GATT & Advertising bearers (proxy.c & adv.c)
 - Network Layer (net.c)
 - Lower and Upper Transport Layers (transport.c)
 - Access Layer (access.c)
 - Foundation Models, Server role (health.c & cfg.c)
 - Both PB-ADV and PB-GATT based provisioning (prov.c)
 - Low Power Node support (lpn.c)
 - Relay support (net.c)
 - GATT Proxy (proxy.c)

Notable features that are *not* part of the implementation:

 - Friend support (initial bits are in place in friend.c)
 - Provisioner support (low-value for typical Zephyr devices)
 - GATT Client (low-value for typical Zephyr devices)

Jira: ZEP-2360

Change-Id: Ic773113dbfd84878ff8cee7fe2bb948f0ace19ed
Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

### CODE REVIEW

**Review by holtmann — APPROVED** [review]

File: subsys/bluetooth/host/mesh/adv.c [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+#include "../hci_core.h"
+
+#include "adv.h"
+#include "foundation.h"
+#include "net.h"
+#include "beacon.h"
+#include "prov.h"
+#include "proxy.h"
+
+#if BT_MESH_LEGACY_DATA_TYPES
```
Comment (Vudentz) [inline subsys/bluetooth/host/mesh/adv.c]: I guess these are not really needed anymore.

Comment (jhedberg) [inline subsys/bluetooth/host/mesh/adv.c]: Good point. Will remove.

Comment (jhedberg) [inline subsys/bluetooth/host/mesh/adv.c]: Removed, and the standard definitions are now added to include/bluetooth/hci.h

**Review by Vudentz — DISMISSED** [review]

**Review by dbkinder — DISMISSED** [review]
doc changes LGTM -- we'll see what doxygen says :)

**Comment by jhedberg** [comment]
Rebased to see if CI would finally complete. (i.e. the "dismissed stale reviews" isn't really valid as nothing changed in the patch)

**Comment by nashif** [comment]
i am looking into the CI issues

**Comment by jhedberg** [comment]
I recreated this pull request since CI seems to have some bugs properly building it. The copy PR can be found here: https://github.com/zephyrproject-rtos/zephyr/pull/850

**Comment by jhedberg** [comment]
The duplicate PR (which CI was able to successfully build) has been merged, so closing this one.

**Pull request closed without merging**